### PR TITLE
Fixed drag and drop in collectionView.

### PR DESCRIPTION
**Pull request by foritari**
Also fixed some other small issues:
- Rename bottom table view section
- Call completion handler on main thread
#6 #5 #4 #2

### CODE REVIEW

File: Image Gallery/ImageGallery/ImageGalleryCollectionViewCell.swift [hunk 1]
```diff
@@ -20,4 +20,9 @@ class ImageGalleryCollectionViewCell: UICollectionViewCell {
         }
     }
     
+    override func prepareForReuse() {
+        imageView.image = nil
```
Comment (kovtun1) [inline Image Gallery/ImageGallery/ImageGalleryCollectionViewCell.swift]: `super`

File: Image Gallery/ImageDetaileView/ImageDetailViewController.swift [hunk 2]
```diff
@@ -29,11 +29,9 @@ class ImageDetailViewController: UIViewController {
     @IBOutlet weak var imageView: UIImageView!
 
     func showImage(url: URL) {
-        imageFetcher?.fetchImage(url: url) { (image) in
-            DispatchQueue.main.async { [weak self] in
-                self?.setupImageView(with: image)
-                self?.activityIndicator.stopAnimating()
-            }
+        imageFetcher?.fetchImage(url: url) { [weak self] (image) in
```
Comment (kovtun1) [inline Image Gallery/ImageDetaileView/ImageDetailViewController.swift]: `strongSelf`

File: Image Gallery/ImageGallery/ImageGallaryViewController.swift [hunk 3]
```diff
@@ -103,13 +108,9 @@ class ImageGallaryViewController: UIViewController, UIDropInteractionDelegate, U
...
             let url = imageGallery.urls[indexPath.item]
-            imageFetcher.fetchImage(url: url as URL) { (image) in
-                DispatchQueue.main.async { [weak self] in
-                    collectionView.performBatchUpdates({
-                        self?.imageGallery.aspectRations[indexPath.item] = image.size.width/image.size.height
-                    })
-                    imageGalleryCell.image = image
-                }
+            imageFetcher.fetchImage(url: url as URL) { [weak self] (image) in
+                self?.imageGallery.aspectRations[indexPath.item] = image.size.width/image.size.height
```
Comment (kovtun1) [inline Image Gallery/ImageGallery/ImageGallaryViewController.swift]: `strongSelf`

Comment (foritari) [reply Image Gallery/ImageGallery/ImageGallaryViewController.swift]: This line was deleted in another branch, that was approved, so we won't have it here too, after merge.

**Pull request merged**